frame/system/src/weights.rs

```diff
@@ -74,3 +65,399 @@ impl ExtrinsicsWeight {
+				&mut error,
+				"[{:?}] {:?} (max block) must fit at least one extrinsic {:?} (base weight)",
+				class, self.max_block, base_extrinsic + self.base_block
+			);
+		}
+
+		if error.has_errors {
+			Err(error)
+		} else {
+			Ok(self)
+		}
+	}
+
+	/// Create new weights definition, with both `Normal` and `Operational`
+	/// classes limited to given weight.
+	///
+	/// Note there is no reservation for `Operational` class, so this constructor
+	/// is not suitable for production deployments.
+	pub fn simple_max(block_weight: Weight) -> Self {
+		Self::builder()
+			.base_block(0)
+			.base_extrinsic(0, ExtrinsicDispatchClass::All)
+			.max_for_non_mandatory(block_weight)
+			.build()
+			.expect("We only specify max_for_class and leave base values as defaults; qed")
+	}
+
+	/// Create a sensible default weights system given only expected maximal block weight and the
+	/// ratio that `Normal` extrinsics should occupy.
+	///
+	/// Assumptions:
+	///  - Average block initialization is assumed to be `10%`.
+	///  - `Operational` transactions have reserved allowance (`1.0 - normal_ratio`)
+	pub fn with_sensible_defaults(
+		expected_block_weight: Weight,
+		normal_ratio: Perbill,
+	) -> Self {
+		let normal_weight = normal_ratio * expected_block_weight;
+		Self::builder()
+			.max_for_class(normal_weight, DispatchClass::Normal)
+			.max_for_class(expected_block_weight, DispatchClass::Operational)
+			.reserved(expected_block_weight - normal_weight, DispatchClass::Operational)
+			.avg_block_initialization(Perbill::from_percent(10))
+			.build()
+			.expect("Sensible defaults are tested to be valid; qed")
+	}
+
+	/// Start constructing new `BlockWeights` object.
+	///
+	/// By default all kinds except of `Mandatory` extrinsics are disallowed.
+	pub fn builder() -> BlockWeightsBuilder {
+		let max_for_class = ForDispatchClass {
+			normal: Some(0),
+			operational: Some(0),
+			mandatory: None,
+		};
+		BlockWeightsBuilder {
+			weights: BlockWeights {
+				base_block: constants::BlockExecutionWeight::get(),
+				base_extrinsic: ForDispatchClass::new(constants::ExtrinsicBaseWeight::get()),
+				max_extrinsic: max_for_class.clone(),
+				reserved: max_for_class.clone(),
+				max_for_class,
+				max_block: 0,
+			},
+			init_cost: Perbill::zero(),
+		}
+	}
+}
+
+/// Dispatch class to apply change to.
+pub enum ExtrinsicDispatchClass {
+	/// All dispatch classes.
+	All,
+	/// Single dispatch class.
+	One(DispatchClass),
+}
+
+impl From<DispatchClass> for ExtrinsicDispatchClass {
+	fn from(c: DispatchClass) -> Self {
+		ExtrinsicDispatchClass::One(c)
+	}
+}
+
+/// An opinionated builder for `Weights` object.
+pub struct BlockWeightsBuilder {
+	weights: BlockWeights,
+	init_cost: Perbill,
+}
+
+impl BlockWeightsBuilder {
+	/// Set base block weight.
+	pub fn base_block(mut self, base_block: Weight) -> Self {
+		self.weights.base_block = base_block;
+		self
+	}
+
+	/// Set base extrinsic weight for given dispatch class.
+	pub fn base_extrinsic(
```

Comment:
```suggestion
	pub fn base_for_class(
```

Seems more consistent with the following functions.